[Galvin Leow Wen Yuan] iP

galvinleow opened the pull request.

Commit:
Add Gradle support

galvinleow committed:
Level 1 - Add Greet, Echo, Exit Feature

galvinleow committed:
Level 2 - Add, List

Edited code to expand scalability:
1. Switch case is used instead of if-else

galvinleow committed:
Level 3 - Mark as Done

Created class for task: 
1. Added feature for done jobs

Refactor print statement for easy of use

galvinleow committed:
Fixed markAsDone Bug

User selected task to be done does not align:
User input and array index different by 1, previous version does not reflect that in code

Add toString method in class file

galvinleow committed:
Change code to fix Level-4 requirement

In level 3, the code used arraylist to get the job done. However, in level 4 using array was the requirement. Hence this commit added code for all functionality till level 3 using array. The arraylist code is commented out. 

Task class changes toString method to better fit future requirements

galvinleow committed:
Change code to fit Level-4 requirement

In level 3, the code used arraylist to get the job done. However, in level 4 using array was the requirement. Hence this commit added code for all functionality till level 3 using array. The arraylist code is commented out. 

Task class changes toString method to better fit future requirements

galvinleow committed:
Merge remote-tracking branch 'origin/master'

galvinleow committed:
Level 4 - ToDo, Event, Deadline

Added 3 new subclass
Each have it own overridden method
Add if-else case to handle scanner for each case

galvinleow committed:
Configure runtest script to run project

Change the javac path to align with project

Refactor nested if-else to switch case

galvinleow committed:
A-TextUiTesting

Handled the UTF-8 encoding issue
Created a test scenario to run test case

galvinleow committed:
Level 5 Handle Errors

Create DukeException Class
Add throw exception at relevant area

galvinleow committed:
Convert task array to task arraylist

All the implementation using task arr is remove.
All the implementation using task arraylist is added, feature up until level 5 only

galvinleow committed:
Level 6 - Delete 

Added Delete feature in switch case
Add test case and change expected outcome

galvinleow committed:
Level 7 Save

Enable save feature

New session every app launch, no task retain.

galvinleow committed:
Level 8 Dates and Times

Add date time feature for event and deadline

String was not very meaningful

galvinleow committed:
Merge branch 'branch-Level-7'

galvinleow committed:
Merge branch 'branch-Level-8'

Solve merge conflict

galvinleow committed:
Fix bug for dateTime integration with storage

Datetime append in wrong format to storage

Fix bug for error when loading file

galvinleow committed:
Refactor to include package

Make Task class and subclass into a package.
Organize the code structure.

Change import statement to not include *.
Change setting on IDE to remove further problems.
Imported class should always be listed explicitly.

galvinleow committed:
Merge branch 'A-Packages'

galvinleow committed:
A-MoreOOP

Create class file to make it more OOP.
This is align with the OOP requirement.

galvinleow committed:
A-MoreOOP

Add delete functionality
Add write to file functionality
Refactor duke class to fit requirement

galvinleow committed:
Create JUnit Test

Create JUnit Test for Task class and subclass
Create JUnit Test for Parser class

galvinleow committed:
Add Junit Test

Add test case for >90% of the code

galvinleow committed:
A-JavaDoc

Add JavaDoc for all class and public method.

galvinleow committed:
A-CodingStandard

Change from 2 space to 4 space.
Previously 2 space as setting set to google codestyle.

galvinleow committed:
Level-9 Find

Add find command.
Add find feature in tasklist class.
Add find option in parser.
Add find feature description for UI.

galvinleow committed:
Merge branch 'branch-Level-9'

galvinleow committed:
Merge branch 'branch-A-JavaDoc'

Resolve merge conflict

galvinleow committed:
Fix merge conflict bug

Missed out some function due to human error.

galvinleow committed:
Add unit test for find function

Ensure that find function is working well.

galvinleow committed:
Format application

Reconfigure formatting rules on intellij.
Better fit with the coding standards.

galvinleow committed:
Merge branch 'add-gradle-support' into A-Gradle

shermanjlim commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,83 @@
+     *
+     * @return String menu
+     */
+    private String menu() {
```
Perhaps getMenu would be a better method name as methods should be named using verbs.

shermanjlim commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,74 @@
+    }
+
+    /**
+     * Main logic of application execution.
```
Maybe the header comments for a method could start with a verb instead of a noun? In the coding standard, they have some suggestions like "Returns", "Sends" etc

shermanjlim commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,74 @@
+package duke;
+
+import duke.command.Command;
+import duke.io.Storage;
+import duke.io.TaskList;
+import duke.parser.Parser;
```
I like how you have structured your application! I think the classes are really well organized in their packages!

shermanjlim commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,83 @@
+public class Ui {
+
+    /**
+     * Print out welcome message with the menu.
```
I am just being nit-picky here but you can consider changing it to "Prints" instead of "Print" as described in the coding standard. I noticed this in several other places too

shermanjlim commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,34 @@
+ * @author A0200204J
+ * @version v1.0
+ */
+public abstract class Command {
```
I like how you are using abstraction here!

shermanjlim commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,123 @@
+        storage.write(taskList);
+    }
+
+    private static LocalDate dateToLocalDate(String[] date) throws DukeException {
```
Maybe this method can be called "convertDateToLocalDate" to sound more like an action! :) Noticed this in other places too!

shermanjlim commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,53 @@
+    public static Command parse(String fullCommand) throws DukeException {
+        String command = fullCommand.split(" ")[0];
+        switch (command) {
+            case "list":
+                return new ListCommand();
+            case "done":
+                return new DoneCommand(fullCommand);
+            case "todo":
+                return new ToDoCommand(fullCommand);
+            case "deadline":
+                return new DeadlineCommand(fullCommand);
+            case "event":
+                return new EventCommand(fullCommand);
+            case "delete":
+                return new DeleteCommand(fullCommand);
+            case "find":
+                return new FindCommand(fullCommand);
+            case "bye":
+                return new ByeCommand();
+            default:
+                throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means");
```
You can consider removing the indentation for the case clauses. I did the same thing too in my codebase previously but it seems like the coding standard is saying that there is no indentation haha :)

joshruien replied:
I agree with shermz-lim, there shouldn't be indentation for case clauses as indicated in the coding standard.

shermanjlim reviewed (commented):
I really like how everything is organized, particularly how the classes are placed in different packages. I think your code is really clean in general too! I was just being nit-picky with some slight deviations from the coding standard. Feel free to check out the comments! 👍

joshruien commented on src/main/java/duke/command/ToDoCommand.java:
```diff
@@ -0,0 +1,59 @@
+ * @author A0200204J
+ * @version v1.0
+ */
+public class ToDoCommand extends Command {
```
Perhaps the class name could be TodoCommand with d in small caps since your todo class has d in small caps. This ill make your code more standardised.

joshruien commented on src/main/java/duke/io/Storage.java:
```diff
@@ -0,0 +1,141 @@
+                String[] tempArr = fileLine.split(",");
+                String command = tempArr[0];
+                switch (command) {
+                    case "duke.task.Todo":
```
Same case here where case clauses shouldn't have indentation.

joshruien commented on src/main/java/duke/io/Storage.java:
```diff
@@ -0,0 +1,141 @@
+                        taskArrayList.add(tempDeadline);
+                        break;
+                    case "duke.task.Event":
+                        Task
```
Should there be a line break here between line 67 and 68? I feel that there is no need for a line break as the line is not long when put together.

joshruien commented on src/test/java/duke/io/DukeIOTest.java:
```diff
@@ -0,0 +1,87 @@
+        String filePath = dataDir + File.separator + dataFileWrite;
+        Storage storage = new Storage(filePath);
+        TaskList taskList = new TaskList(storage.load());
+        // Create Test Result to compare
```
Perhaps there could be a line break here to improve readability of your code.

joshruien reviewed (commented):
LGTM, code is quite clean, just a few things to change but overall good job!  👍

galvinleow committed:
Fixed code style

Use checkstyle to check code style violation and resolve issue.

galvinleow committed:
Merge branch 'A-Gradle'

galvinleow committed:
Change to fxml format

galvinleow committed:
Refactor code to to use UI class to print

galvinleow committed:
Create link between duke and javaFx

This commit removes some of the method used previously so becareful.

galvinleow committed:
Merge branch 'branch-Level-10'

galvinleow committed:
Edit code based on PR review

Edit javaDoc description to fit java coding standard

galvinleow committed:
Disable some test case

Test case not updated
Wait for feature done to build test case

galvinleow committed:
Update main class for gradle

Launch from the correct file.
Change so that jar file will be correct.

galvinleow committed:
Add assertion

Add assert for program to meet requirements

galvinleow committed:
Improve code quality

Better naming convention so that it can be easily identifiable

galvinleow committed:
Merge pull request #1 from galvinleow/branch-A-Assertions

Add assertion

galvinleow committed:
Merge branch 'branch-A-CodeQuality'

galvinleow committed:
Create update feature

User can update the description or the date for the task so that if mistake were made they can change the value instead of deleting and recreating.

galvinleow committed:
Set theme jekyll-theme-midnight

galvinleow committed:
Update readme

Create user guide in docs readme

galvinleow committed:
Merge branch 'branch-A-UserGuide'

galvinleow committed:
Update UG

Component not showing properly

galvinleow committed:
Update Dialog Box

Add color to conversion so that it will have some contrast to the background

galvinleow committed:
Fix bug

Data not writing to txt file when update desc so it is need to be fixed

Enable case insensitive in find function

galvinleow committed:
Add image  dashboard for iP showcase
